lib/model.dart

```diff
@@ -67,14 +63,12 @@ class Model {
 class Item {
   const Item({
     required this.bounds,
-    this.selected = false,
```

Comment:
I was wondering whether being selected is part of the model or if it should just be driven by what has the focus.

At first it seemed strange that there are two independent types of "selection": what has focus and what is selected in the model. However, the more I think about it, maybe those two need to be separate. I am thinking of a use case where we also show e.g. a dropdown menu to change the color of an Item. In that case, you want to select an item by clicking on it and then you move the focus to the drop down to select a different color for it - throughout this entire time, the item should probably be drawn as selected even though it does not have focus.

If we split up the concept of focus and selected, I am further wondering when you should be able to move a selected item with the keyboard. Only when it also has focus? Or can I move it while focus is somewhere else (say in said color picker)?.

What do people think?

Reply:
I tried this in the google docs drawing editor. It differentiates between "has focus" and "selected". However, the keyboard shortcuts to move something only work if the focus is inside the drawing area (e.g. while focus is on the "Save and Close" button of the window, you cannot move a rect in the drawing area even when it is selected). Should we go with those semantics for our demo here as well?

Reply:
Yes, that makes sense to me, although the google docs example is further complicated by being a web app.  Other authoring programs share this model too, but most of them only move focus away from the canvas when there is a popup menu or dialog open.

Reply:
I agree that the focus and the current selection are different. The selection could be five items and one of those items might have an editable textfield stacked on top of its label because the user is editing its label. Or driving the focus around within a popup for selecting the item's background color.

To handle shortcuts the app or the app's window needs to have the focus but I don't think should have anything to do with the list of items that's selected within the app.

Reply:
I've changed the implementation to treat focus and selection separately.